src/test/java/com/buddy/api/units/domains/services/impls/CreateAccountServiceTest.java

```diff
@@ -0,0 +1,108 @@
+import com.buddy.api.domains.account.dtos.AccountDto;
+import com.buddy.api.domains.account.entities.AccountEntity;
+import com.buddy.api.domains.account.repository.AccountRepository;
+import com.buddy.api.domains.account.services.impl.CreateAccountServiceImpl;
+import com.buddy.api.units.UnitTestAbstract;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+public class CreateAccountServiceTest extends UnitTestAbstract {
+    @Mock
+    private AccountRepository accountRepository;
+
+    @Mock
+    private PasswordEncoder passwordEncoder;
+
+    @InjectMocks
+    CreateAccountServiceImpl createAccountService;
+
+    private final Boolean termsOfUserConsent = true;
+
+    @Test
+    @DisplayName("Should create account")
+    void should_create_account() {
+        final var email = generateValidEmail();
+        final var phoneNumber = generateRandomPhoneNumber();
+        final var password = generateRandomPassword();
+        final var encryptedPassword = generateRandomPassword();
+
+        final AccountDto accountDto = new AccountDto(
+            email,
+            phoneNumber,
+            password,
+            termsOfUserConsent
+        );
+
+        final AccountEntity accountEntity = new AccountEntity(
+            null,
+            email,
+            phoneNumber,
+            encryptedPassword,
+            termsOfUserConsent,
+            false,
+            false,
+            false,
+            null,
+            null,
+            null
+        );
+
+        when(accountRepository.findByEmail(email)).thenReturn(Optional.empty());
+        when(passwordEncoder.encode(password)).thenReturn(encryptedPassword);
+        when(accountRepository.save(accountEntity)).thenReturn(accountEntity);
+
+        createAccountService.create(accountDto);
+
+        verify(passwordEncoder, times(1)).encode(password);
+        verify(accountRepository, times(1)).save(accountEntity);
+    }
+
+    @Test
+    @DisplayName("Should not create account when email is already in database")
+    void should_not_create_account_when_email_is_already_in_database() {
+        final var email = generateValidEmail();
+
+        final AccountEntity accountEntity = validAccountEntity()
+            .accountId(UUID.randomUUID())
+            .creationDate(LocalDateTime.now())
+            .updatedDate(LocalDateTime.now())
+            .build();
+
+        final AccountDto accountDto = new AccountDto(
+            email,
+            generateRandomPhoneNumber(),
+            generateRandomPassword(),
+            termsOfUserConsent
+        );
+
+        when(accountRepository.findByEmail(email)).thenReturn(Optional.of(accountEntity));
+
+        assertThatThrownBy(() -> createAccountService.create(accountDto))
```

Comment:
Sugestão
```suggestion
       assertThatThrownBy(() -> createAccountService.create(accountDto))
            .isInstanceOf(EmailAlreadyRegisteredException.class)
            .hasMessage("Account email already registered")
            .extracting("fieldName")
            .isEqualTo("email");

        verify(accountRepository, never()).save(accountEntity);
```

Reply:
```suggestion
        verify(accountRepository, never()).save(accountEntity);
```

Nessa parte acredito que o `any()` seja mais apropriado pois eu não quero validar que ele não salvou `accountEntity` especificamente neste caso. Se por exemplo, um campo de `accountEntity` for modificado dentro do teste, ele vai validar que realmente não tentou salvar `accountEntity` (porque lembra que não tá verificando por instância aqui, já que dentro do service, é criado uma nova instância), e ainda assim, uma entidade estaria sendo salva no banco de dados.

De fato, o que esse teste tenta garantir que neste cenário o service não irá chamar o repository para tentar salvar `nenhuma` `AccountEntity`, pois ele só deve fazer isso para a `AccountEntity` informada e somente se ela for válida.

Reply:
Entendendo o que você pensou sobre isso, acredito que é a solução mais correta para fazer uso do **any()**, porque o que estamos testando aqui é garantir que o método `save()` **não** seja chamado, independentemente da instância exata da `AccountEntity`. Isso faz sentido, pois a instância criada dentro do serviço pode ser diferente da que passamos no teste, mas o comportamento esperado (não salvar nada) deve ser o mesmo. 

Em relação ao `when()`, também acho que o uso de `any()` no `findByEmail()` é válido aqui. O que realmente importa é que, quando o e-mail já existe no banco, o serviço não deve tentar salvar nenhuma conta, e não tanto a string exata passada para o método.

Por exemplo, isso aqui ficaria legal:

```java
when(accountRepository.findByEmail(any(String.class))).thenReturn(Optional.of(accountEntity));
verify(accountRepository, never()).save(any(AccountEntity.class));
```

No final das contas, o uso do `any()` vai garantir que estamos validando o comportamento certo: que, quando o e-mail já está registrado, o `save()` nunca deve ser chamado, sem se preocupar com os detalhes da instância da entidade.

Acho que essa abordagem deixa o teste mais flexível e menos propenso a falhar por questões de implementação e concordo com a decisão. Sobre o que foi sugerido para incrementar o que acha?